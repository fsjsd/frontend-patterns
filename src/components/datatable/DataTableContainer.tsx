import React, { Suspense } from 'react'
import { ContentContainer } from '../../ux/ContentContainer'
import Loading from '../../ux/Loading'
const DataTable = React.lazy(() => import('./DataTable'))

const DataTableContainer = () => {
  return (
    <ContentContainer
      title="Data Table"
      codeLink="/features/datatable"
      markDownPromise={import('./requirements.md')}
    >
      <Suspense fallback={<Loading />}>
        <div role="main">
          <DataTable />
        </div>
      </Suspense>
    </ContentContainer>
  )
}

export default DataTableContainer