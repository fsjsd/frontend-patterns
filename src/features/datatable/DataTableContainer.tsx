import React, { Suspense } from 'react'
import ContentHeader, { ContentHeaderLabel } from '../../ux/ContentHeader'
import { ContentWrapper } from '../../ux/ContentWrapper'
import Loading from '../../ux/Loading'
const DataTable = React.lazy(() => import('./DataTable'))

const DataTableContainer = () => {
  return (<>
    <ContentHeader>
      <ContentHeaderLabel>Data Table</ContentHeaderLabel>
    </ContentHeader>
    <ContentWrapper>
      <Suspense fallback={<Loading />}>
        <div role="main">
          <DataTable />
        </div>
      </Suspense>
    </ContentWrapper>
  </>
  )
}

export default DataTableContainer