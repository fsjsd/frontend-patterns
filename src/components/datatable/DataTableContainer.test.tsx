import React from 'react'
import DataTableContainer from "./DataTableContainer";
import { render, waitFor } from '@testing-library/react'

describe("Data Table", () => {
  it("renders and matches snapshot", async () => {
    const { container, queryByRole } = render(<DataTableContainer />)
    await waitFor(() => expect(queryByRole("main")).toBeInTheDocument());
    expect(container).toMatchSnapshot();

  })
})