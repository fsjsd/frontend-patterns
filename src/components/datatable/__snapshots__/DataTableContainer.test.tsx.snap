// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table renders and matches snapshot 1`] = `
<div>
  <header
    class="sc-bczRLJ dmSzxI"
  >
    <div
      class="sc-gsnTZi hCGKc"
      role="complementary"
    >
      Data Table
    </div>
    <div
      class="sc-dkzDqf gedmVz"
    >
      <div
        class="sc-hKMtZM dNwpDv"
        role="button"
        title="Run demo"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M10 16.5l6-4.5-6-4.5zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="sc-hKMtZM itnQjY"
        role="button"
        title="View notes"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M3 18h12v-2H3v2zM3 6v2h18V6H3zm0 7h18v-2H3v2z"
          />
        </svg>
      </div>
      <a
        aria-label="View source code on GitHub"
        href="https://github.com/fsjsd/frontend-patterns/tree/main/src/features/datatable"
        rel="noopener noreferrer"
        role="button"
        target="_blank"
      >
        <div
          class="sc-hKMtZM itnQjY"
          title="View code on github"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </a>
    </div>
  </header>
  <section
    class="sc-jSMfEi ljxiYZ"
    role="document"
  >
    <div
      role="main"
      style=""
    >
      <div
        class="sc-iBkjds fPMrdA"
      >
        Coming soon
      </div>
    </div>
  </section>
</div>
`;
